pkg/orchestrator/etcd_worker_test.go

```diff
@@ -0,0 +1,266 @@
+	if newState == nil {
+		return s.state, nil
+	}
+	s.state = newState
+
+	if s.id == 0 {
+		sum := s.state.sum
+		for _, delta := range s.state.deltas {
+			sum = sum - delta.old
+			sum = sum + delta.new
+		}
+
+		// check for consistency
+		expectedSum := 0
+		for i := range s.state.values {
+			for j := range s.state.values[i] {
+				expectedSum += s.state.values[i][j]
+			}
+		}
+		if sum != expectedSum {
+			log.Panic("state is inconsistent", zap.Int("expected-sum", sum), zap.Int("actual-sum", s.state.sum))
+		}
+
+		s.state.SetSum(sum)
+	} else {
+		i2 := s.id - 1
+		for i := range s.state.values {
+			s.state.Inc(i, i2)
+		}
+	}
+
+	s.state.deltas = s.state.deltas[:0]
+
+	return s.state, nil
+}
+
+type delta struct {
+	old int
+	new int
+	i1  int
+	i2  int
+}
+
+type simpleReactorState struct {
+	values  [][]int
+	sum     int
+	deltas  []*delta
+	patches []*DataPatch
+}
+
+var (
+	keyParseRegexp = regexp.MustCompile(`/(.+)`)
+)
+
+func (s *simpleReactorState) Get(i1, i2 int) int {
+	return s.values[i1][i2]
+}
+
+func (s *simpleReactorState) Inc(i1, i2 int) {
+	patch := &DataPatch{
+		Key: []byte("/" + strconv.Itoa(i1)),
+		Fun: func(old []byte) ([]byte, error) {
+			var oldJSON []int
+			err := json.Unmarshal(old, &oldJSON)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+
+			oldJSON[i2]++
+			return json.Marshal(oldJSON)
+		},
+	}
+
+	s.patches = append(s.patches, patch)
+}
+
+func (s *simpleReactorState) SetSum(sum int) {
+	patch := &DataPatch{
+		Key: []byte("/sum"),
+		Fun: func(_ []byte) ([]byte, error) {
+			return []byte(strconv.Itoa(sum)), nil
+		},
+	}
+
+	s.patches = append(s.patches, patch)
+}
+
+func (s *simpleReactorState) Update(key []byte, value []byte) {
+	subMatches := keyParseRegexp.FindSubmatch(key)
+	if len(subMatches) != 2 {
+		log.Panic("illegal Etcd key", zap.ByteString("key", key))
+	}
+
+	if string(subMatches[1]) == "sum" {
+		newSum, err := strconv.Atoi(string(value))
+		if err != nil {
+			log.Panic("illegal sum", zap.Error(err))
+		}
+		s.sum = newSum
+		return
+	}
+
+	index, err := strconv.Atoi(string(subMatches[1]))
+	if err != nil {
+		log.Panic("illegal index", zap.Error(err))
+	}
+
+	var newValues []int
+	err = json.Unmarshal(value, &newValues)
+	if err != nil {
+		log.Panic("illegal value", zap.Error(err))
+	}
+
+	for i2, v := range s.values[index] {
+		if v != newValues[i2] {
+			s.deltas = append(s.deltas, &delta{
+				old: v,
+				new: newValues[i2],
+				i1:  index,
+				i2:  i2,
+			})
+		}
+	}
+
+	s.values[index] = newValues
+}
+
+func (s *simpleReactorState) GetPatches() []*DataPatch {
+	ret := s.patches
+	s.patches = nil
+	return ret
+}
+
+func (s *etcdWorkerSuite) SetUpTest(c *check.C) {
+	dir := c.MkDir()
+	url, etcdServer, err := etcd.SetupEmbedEtcd(dir)
+	c.Assert(err, check.IsNil)
+	s.etcdServer = etcdServer
+	s.endpoints = []string{url.String()}
+
+	log.Debug("Set up embedded etcd", zap.Strings("endpoints", s.endpoints))
+}
+
+func (s *etcdWorkerSuite) TestEtcdSum(c *check.C) {
```

Comment:
Could you test some anomalies against to etcd guarantee? To make sure CDC is comfortable with  some corner cases.